build-scripts/build-custom-app.sh

```diff
@@ -0,0 +1,251 @@
+
+type TProps = {
+  children?: ReactNode;
+};
+
+const Main: FC<TProps> = ({ children }) => (
+  <>
+    <Header />
+    <main id="main" className="flex flex-col flex-1">
+      {children}
+    </main>
+    <Footer />
+  </>
+);
+export default Main;
+
+EOM
+
+# Define the content for Analytics.tsx
+read -r -d '' ANALYTICS_CONTENT <<EOM
+
+import React from 'react';
+
+const Analytics = ({enableAnalytics}: {enableAnalytics: boolean}) => {
+    return null 
+};
+
+export default Analytics;
+
+EOM
+
+# Define the content for methodology.tsx
+read -r -d '' METHODOLOGY_PAGE_CONTENT <<EOM
+
+const Methodology = () => {
+  return null;
+};
+
+export default Methodology;
+
+EOM
+
+# Define the content for homepage.tsx
+read -r -d '' HOMEPAGE_CONTENT <<EOM
+
+import React from "react";
+// generic layer component
+import Layout from "@components/layouts/LandingPage";
+
+import Hero from "@${theme}/components/Hero";
+
+type TProps = {
+  handleSearchInput: (term: string, filter?: string, filterValue?: string) => void;
+  searchInput: string;
+};
+
+const LandingPage = ({ handleSearchInput, searchInput }: TProps) => {
+  return (
+    <Layout title="">
+      <main id="main" className="flex flex-col flex-1">
+        <div className="bg-cclw-dark">
+          <Hero handleSearchInput={handleSearchInput} searchInput={searchInput} />
+        </div>
+      </main>
+    </Layout>
+  );
+};
+
+export default LandingPage;
+
+
+EOM
+
+# Define the content for the hero.
+
+read -r -d '' HERO_CONTENT <<EOM
+
+import LandingSearchForm from "./LandingSearchForm";
+
+
+type TProps = {
+  handleSearchInput: (term: string, filter?: string, filterValue?: string) => void;
+  searchInput: string;
+};
+
+const Hero = ({ handleSearchInput, searchInput }: TProps) => {
+  return (
+    <div className="pb-6 text-white pt-[28px] sm:pt-[48px] md:pt-[80px] lg:pt-[100px] xl:pt-[140px]">
+      <div className="container">
+        <div className="flex flex-col items-center justify-center mb-6">
+		  <p className="font-medium tracking-slight text-lg lg:text-3xl" data-cy="intro-message">
+          ${custom_app_name} Custom App
+        </p>
+        </div>
+        <div className="max-w-screen-md mx-auto mt-6">
+          <LandingSearchForm handleSearchInput={handleSearchInput} placeholder="Search full text of any document" input={searchInput} />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default Hero;
+
+
+EOM
+
+# Define the content for MethodologyLink.tsx
+read -r -d '' METHODOLOGY_LINK_CONTENT <<EOM
+import { LinkWithQuery } from "@components/LinkWithQuery";
+
+const MethodologyLink = () => <LinkWithQuery href="/methodology">our methodology page</LinkWithQuery>;
+
+export default MethodologyLink;
+EOM
+
+# Define the content for the routes.json
+
+read -r -d '' ROUTES_JSON_CONTENT <<EOM
+
+[
+  {
+    "title": "Methodology",
+    "path": "methodology",
+    "contentPath": "methodology"
+  }
+]
+
+EOM
+
+# Define the content for the index.ts file
```

Comment:
The index.ts file for what?

Reply:
this should function as the primary access point for components in this directory (bit like the init.py), will update

Reply:
Why is this needed also?